internal/connect/zypper.go

```diff
@@ -199,3 +203,31 @@ func setReleaseVersion(version string) error {
 	_, err := zypperRun(args, true, []int{zypperOK})
 	return err
 }
+
+// RefreshRepos runs zypper to refresh all repositories
+func RefreshRepos(version string, force bool, quiet bool, verbose bool, nonInteractive bool) error {
+	args := []string{"ref"}
+	flags := []string{}
+	if force {
+		args = append(args, "-f")
+	}
+	if nonInteractive {
+		flags = append(flags, "--non-interactive")
+	}
+	if verbose {
+		flags = append(flags, "--verbose")
+	}
+	if quiet {
+		flags = append(flags, "--quiet")
+	}
+	if version != "" {
+		flags = append(flags, "--releasever", version)
+	}
+	args = append(flags, args...)
+	output, err := zypperRun(args, quiet, []int{zypperOK})
+	// TODO: print output as it goes not post-factum and do it for all zypper calls!
+	if len(output) > 0 {
+		fmt.Print(string(output))
```

Comment:
should this be the quiet logger instead of `fmt`?

Reply:
this should not be needed at all. we have to find some way to nicely print the output as the comment says. this is very important for long runs like `zypper dup`.

Reply:
Yes, that's exactly what we need for zypper-migration-plugin. We might need to extend the current execute logic or make it somehow "pluggable" with different "executors".

Reply:
Okay so the Ruby zypper migration plugin uses `system` to directly call zypper in several places, so that output is displays as it comes from zypper.

My preference would be to create a new function like zypperLongRun or something and use it for these cases. I'm not keen about doing clever stuff that makes the current zypperRun/execute() more complex

Reply:
Maybe we could use the new `execute()` func variable to change this functionality in zypper-migration case? Only two `snapper` calls use the backticks to grab output from the command. The rest is called via `system`. I'm not sure about the commands called inside connect package.

Reply:
Or maybe we could just change the existing execute() to always use pipes, read the pipe, print (if asked) and accumulate the output in a buffer as it comes. Then return the output at the end.